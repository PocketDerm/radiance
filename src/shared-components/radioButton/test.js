import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import RadioButton from './index';

describe('<RadioButton />', () => {
  describe('UI snapshots', () => {
    test('when children is undefined', () => {
      const tree = renderer
        .create(<RadioButton checked={false} onClick={() => {}} />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    test('when children is a node', () => {
      const tree = renderer
        .create(
          <RadioButton checked={false} onClick={() => {}}>
            RadioButton Text
          </RadioButton>
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    // TODO test for primary and secondary type
  });

  describe('onClick callback', () => {
    it('is invoked on click', () => {
      const spy = jest.fn();
      const wrapper = shallow(<RadioButton checked={false} onClick={spy} />);

      wrapper.simulate('click');
      expect(spy).toHaveBeenCalled();
    });
  });
});
