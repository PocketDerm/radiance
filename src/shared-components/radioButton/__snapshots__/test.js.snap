// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> UI snapshots when checked type is primary 1`] = `
.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  cursor: pointer;
  margin-bottom: 20px;
}

.emotion-3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-5 {
  margin-top: 0.25rem;
  margin-left: 1rem;
  min-width: 125px;
}

.emotion-0 {
  color: #ffffff;
  position: absolute;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  top: 50%;
  left: 50%;
  fill: currentColor;
  cursor: pointer;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 2rem;
  width: 2rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 2px solid;
  cursor: pointer;
  border-radius: 100%;
  -webkit-transition: background-color 350ms;
  transition: background-color 350ms;
  background-color: #332e54;
  border-color: #332e54;
}

.emotion-1:active,
.emotion-1:focus {
  outline: none;
}

<div
  className="emotion-7 emotion-8"
  onClick={[Function]}
>
  <div
    className="emotion-3 emotion-4"
  >
    <<svg viewBox="0 0 18 13" fill="currentColor" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
  <title>Done</title>
  <path d="M6.48528137,12.0710678 L1.70710678,7.29289322 C1.31658249,6.90236893 0.683417511,6.90236893 0.292893219,7.29289322 C-0.0976310729,7.68341751 -0.0976310729,8.31658249 0.292893219,8.70710678 L5.77817459,14.1923882 C6.16869888,14.5829124 6.80186386,14.5829124 7.19238816,14.1923882 L17.6776695,3.70710678 C18.0681938,3.31658249 18.0681938,2.68341751 17.6776695,2.29289322 C17.2871452,1.90236893 16.6539803,1.90236893 16.263456,2.29289322 L6.48528137,12.0710678 Z"/>
</svg>

      className="emotion-0"
      height={11}
      width={11}
    />
    <div
      checked={true}
      className="emotion-1 emotion-2"
      type="primary"
    />
  </div>
  <p
    className="emotion-5 emotion-6"
  >
    RadioButton Text
  </p>
</div>
`;

exports[`<RadioButton /> UI snapshots when checked type is secondary 1`] = `
.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  cursor: pointer;
  margin-bottom: 20px;
}

.emotion-3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-5 {
  margin-top: 0.25rem;
  margin-left: 1rem;
  min-width: 125px;
}

.emotion-0 {
  color: #ffffff;
  position: absolute;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  top: 50%;
  left: 50%;
  fill: currentColor;
  cursor: pointer;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 2rem;
  width: 2rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 2px solid;
  cursor: pointer;
  border-radius: 100%;
  -webkit-transition: background-color 350ms;
  transition: background-color 350ms;
  background-color: #a6a1e2;
  border-color: #a6a1e2;
}

.emotion-1:active,
.emotion-1:focus {
  outline: none;
}

<div
  className="emotion-7 emotion-8"
  onClick={[Function]}
>
  <div
    className="emotion-3 emotion-4"
  >
    <<svg viewBox="0 0 18 13" fill="currentColor" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
  <title>Done</title>
  <path d="M6.48528137,12.0710678 L1.70710678,7.29289322 C1.31658249,6.90236893 0.683417511,6.90236893 0.292893219,7.29289322 C-0.0976310729,7.68341751 -0.0976310729,8.31658249 0.292893219,8.70710678 L5.77817459,14.1923882 C6.16869888,14.5829124 6.80186386,14.5829124 7.19238816,14.1923882 L17.6776695,3.70710678 C18.0681938,3.31658249 18.0681938,2.68341751 17.6776695,2.29289322 C17.2871452,1.90236893 16.6539803,1.90236893 16.263456,2.29289322 L6.48528137,12.0710678 Z"/>
</svg>

      className="emotion-0"
      height={11}
      width={11}
    />
    <div
      checked={true}
      className="emotion-1 emotion-2"
      type="secondary"
    />
  </div>
  <p
    className="emotion-5 emotion-6"
  >
    RadioButton Text
  </p>
</div>
`;

exports[`<RadioButton /> UI snapshots when children is a node 1`] = `
.emotion-0 {
  color: transparent;
  position: absolute;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  top: 50%;
  left: 50%;
  fill: currentColor;
  cursor: pointer;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  cursor: pointer;
  margin-bottom: 20px;
}

.emotion-3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 2rem;
  width: 2rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 2px solid;
  cursor: pointer;
  border-radius: 100%;
  -webkit-transition: background-color 350ms;
  transition: background-color 350ms;
  background-color: transparent;
  border-color: #332e54;
}

.emotion-1:active,
.emotion-1:focus {
  outline: none;
}

.emotion-5 {
  margin-top: 0.25rem;
  margin-left: 1rem;
  min-width: 125px;
}

<div
  className="emotion-7 emotion-8"
  onClick={[Function]}
>
  <div
    className="emotion-3 emotion-4"
  >
    <<svg viewBox="0 0 18 13" fill="currentColor" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
  <title>Done</title>
  <path d="M6.48528137,12.0710678 L1.70710678,7.29289322 C1.31658249,6.90236893 0.683417511,6.90236893 0.292893219,7.29289322 C-0.0976310729,7.68341751 -0.0976310729,8.31658249 0.292893219,8.70710678 L5.77817459,14.1923882 C6.16869888,14.5829124 6.80186386,14.5829124 7.19238816,14.1923882 L17.6776695,3.70710678 C18.0681938,3.31658249 18.0681938,2.68341751 17.6776695,2.29289322 C17.2871452,1.90236893 16.6539803,1.90236893 16.263456,2.29289322 L6.48528137,12.0710678 Z"/>
</svg>

      className="emotion-0"
      height={11}
      width={11}
    />
    <div
      checked={false}
      className="emotion-1 emotion-2"
      type="primary"
    />
  </div>
  <p
    className="emotion-5 emotion-6"
  >
    RadioButton Text
  </p>
</div>
`;

exports[`<RadioButton /> UI snapshots when children is undefined 1`] = `
.emotion-0 {
  color: transparent;
  position: absolute;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  top: 50%;
  left: 50%;
  fill: currentColor;
  cursor: pointer;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  cursor: pointer;
  margin-bottom: 20px;
}

.emotion-3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 2rem;
  width: 2rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 2px solid;
  cursor: pointer;
  border-radius: 100%;
  -webkit-transition: background-color 350ms;
  transition: background-color 350ms;
  background-color: transparent;
  border-color: #332e54;
}

.emotion-1:active,
.emotion-1:focus {
  outline: none;
}

<div
  className="emotion-5 emotion-6"
  onClick={[Function]}
>
  <div
    className="emotion-3 emotion-4"
  >
    <<svg viewBox="0 0 18 13" fill="currentColor" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
  <title>Done</title>
  <path d="M6.48528137,12.0710678 L1.70710678,7.29289322 C1.31658249,6.90236893 0.683417511,6.90236893 0.292893219,7.29289322 C-0.0976310729,7.68341751 -0.0976310729,8.31658249 0.292893219,8.70710678 L5.77817459,14.1923882 C6.16869888,14.5829124 6.80186386,14.5829124 7.19238816,14.1923882 L17.6776695,3.70710678 C18.0681938,3.31658249 18.0681938,2.68341751 17.6776695,2.29289322 C17.2871452,1.90236893 16.6539803,1.90236893 16.263456,2.29289322 L6.48528137,12.0710678 Z"/>
</svg>

      className="emotion-0"
      height={11}
      width={11}
    />
    <div
      checked={false}
      className="emotion-1 emotion-2"
      type="primary"
    />
  </div>
</div>
`;

exports[`<RadioButton /> UI snapshots when type is primary 1`] = `
.emotion-0 {
  color: transparent;
  position: absolute;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  top: 50%;
  left: 50%;
  fill: currentColor;
  cursor: pointer;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  cursor: pointer;
  margin-bottom: 20px;
}

.emotion-3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 2rem;
  width: 2rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 2px solid;
  cursor: pointer;
  border-radius: 100%;
  -webkit-transition: background-color 350ms;
  transition: background-color 350ms;
  background-color: transparent;
  border-color: #332e54;
}

.emotion-1:active,
.emotion-1:focus {
  outline: none;
}

.emotion-5 {
  margin-top: 0.25rem;
  margin-left: 1rem;
  min-width: 125px;
}

<div
  className="emotion-7 emotion-8"
  onClick={[Function]}
>
  <div
    className="emotion-3 emotion-4"
  >
    <<svg viewBox="0 0 18 13" fill="currentColor" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
  <title>Done</title>
  <path d="M6.48528137,12.0710678 L1.70710678,7.29289322 C1.31658249,6.90236893 0.683417511,6.90236893 0.292893219,7.29289322 C-0.0976310729,7.68341751 -0.0976310729,8.31658249 0.292893219,8.70710678 L5.77817459,14.1923882 C6.16869888,14.5829124 6.80186386,14.5829124 7.19238816,14.1923882 L17.6776695,3.70710678 C18.0681938,3.31658249 18.0681938,2.68341751 17.6776695,2.29289322 C17.2871452,1.90236893 16.6539803,1.90236893 16.263456,2.29289322 L6.48528137,12.0710678 Z"/>
</svg>

      className="emotion-0"
      height={11}
      width={11}
    />
    <div
      checked={false}
      className="emotion-1 emotion-2"
      type="primary"
    />
  </div>
  <p
    className="emotion-5 emotion-6"
  >
    RadioButton Text
  </p>
</div>
`;

exports[`<RadioButton /> UI snapshots when type is secondary 1`] = `
.emotion-0 {
  color: transparent;
  position: absolute;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  top: 50%;
  left: 50%;
  fill: currentColor;
  cursor: pointer;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  cursor: pointer;
  margin-bottom: 20px;
}

.emotion-3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 2rem;
  width: 2rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 2px solid;
  cursor: pointer;
  border-radius: 100%;
  -webkit-transition: background-color 350ms;
  transition: background-color 350ms;
  background-color: transparent;
  border-color: #332e54;
}

.emotion-1:active,
.emotion-1:focus {
  outline: none;
}

.emotion-5 {
  margin-top: 0.25rem;
  margin-left: 1rem;
  min-width: 125px;
}

<div
  className="emotion-7 emotion-8"
  onClick={[Function]}
>
  <div
    className="emotion-3 emotion-4"
  >
    <<svg viewBox="0 0 18 13" fill="currentColor" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
  <title>Done</title>
  <path d="M6.48528137,12.0710678 L1.70710678,7.29289322 C1.31658249,6.90236893 0.683417511,6.90236893 0.292893219,7.29289322 C-0.0976310729,7.68341751 -0.0976310729,8.31658249 0.292893219,8.70710678 L5.77817459,14.1923882 C6.16869888,14.5829124 6.80186386,14.5829124 7.19238816,14.1923882 L17.6776695,3.70710678 C18.0681938,3.31658249 18.0681938,2.68341751 17.6776695,2.29289322 C17.2871452,1.90236893 16.6539803,1.90236893 16.263456,2.29289322 L6.48528137,12.0710678 Z"/>
</svg>

      className="emotion-0"
      height={11}
      width={11}
    />
    <div
      checked={false}
      className="emotion-1 emotion-2"
      type="secondary"
    />
  </div>
  <p
    className="emotion-5 emotion-6"
  >
    RadioButton Text
  </p>
</div>
`;
